### ISSUE TITLE
Bit-* operations don't work on multidimensional bitarrays

### ISSUE THREAD

**Issue by kpoeck** [clasp-developers/clasp#494]
````
COMMON-LISP-USER> (bit-and (make-array (list 2 2) :element-type 'bit :initial-element 0) (make-array (list 2 2) :element-type 'bit :initial-element 1))
Condition of type: SIMPLE-PROGRAM-ERROR
Illegal arguments for bit-array operation.
````

**Comment by kpoeck** [comment]
Obviously failing somewhere in 
CL_DEFUN T_sp core__bit_array_op(int opval, T_sp tx, T_sp ty, T_sp tr)
in bits.cc

**Comment by kpoeck** [comment]
The error is here:
void asAbstractSimpleVectorRange(AbstractSimpleVector_sp& sv, size_t& start, size_t& end) const final {
...
end = this->length()+this->_DisplacedIndexOffset;

Since length is initalised with 0xDEADBEEF01234567.

Probably length() should throw an error for MD-arrays

**Comment by kpoeck** [comment]
Should mostly be fixed by pr #500 (but 1 test (hell-mdbit-3) is still giving wrong results

**Comment by kpoeck** [comment]
I also fixed the problem with hell-mdbit-3, it is core__bit_array_op where it should read 
````
....
 if (tr == _lisp->_true())
    tr = tx;
....
````
intead of tr = x;

Will send another pr  (or update my existing)
